api/app/tag.py

```diff
@@ -196,19 +198,33 @@ async def fetch_metadata(iss: str, product: str):
             code="failed_to_fetch_metadata",
         )
 
+    try:
+        config = await get_dataspace_configuration(product_passport.product_dataspace)
+        gateway = config["product_gateway_url"]
+    except KeyError:
+        logger.exception(
+            f"product_gateway_url not found in {product_passport.product_dataspace} configuration")
+        raise TagsError(
+            error="Product gateway URL not found in dataspace configuration.",
+            code="failed_to_fetch_metadata",
+        )
```

Comment:
```suggestion
    config = await get_dataspace_configuration(product_passport.product_dataspace)
    gateway = config["product_gateway_url"]
```
This is not a common error case we want to be reporting to end users. It should just crash. Not every potential exception needs error handling logic and custom error messages.